### ISSUE TITLE
no-useless-spread  false positive when spreading a Set()

### ISSUE THREAD

**Issue by jelly** [oxc-project/oxc#4980]
The following warning has an fixer available which transforms the code into
```
  ⚠ eslint-plugin-unicorn(no-useless-spread): Using a spread operator here creates a new array unnecessarily.
     ╭─[src/helpers.js:703:19]
 702 │
 703 │     const uniq = [...new Set(devs)];
     ·                   ───
 704 │     uniq.sort();
     ╰────
  help: `new Set(devs)` returns a new array. Spreading it into an array expression to create a new array is redundant.
```

```
-    const uniq = [...new Set(devs)];
+    const uniq = [new Set(devs)];
```

This would fails as a `Set` has no `sort()`, but maybe more importantly the spread operator isn't useless as removing it produces different code as can be seen below:
```
> new Set([1,2,3,3]).sort()
Uncaught TypeError: (intermediate value).sort is not a function
> [...new Set([1,2,3,3])].sort()
[ 1, 2, 3 ]
> [new Set([1,2,3,3])].sort()
[ Set(3) { 1, 2, 3 } ]
```

**Comment by camc314** [comment]
duplicate of https://github.com/oxc-project/oxc/issues/4872 ? can you confirm?

**Comment by jelly** [comment]
> duplicate of #4872 ? can you confirm?

Sorry, should have tested `main`. This is indeed fixed.

**Comment by camc314** [comment]
> > duplicate of #4872 ? can you confirm?
> 
> 
> 
> Sorry, should have tested `main`. This is indeed fixed.

no worries, thanks for verifying
